pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/blocks/TransferableBlockUtils.java

```diff
@@ -43,4 +48,61 @@ public static TransferableBlock getErrorTransferableBlock(Map<Integer, String> e
   public static boolean isEndOfStream(TransferableBlock transferableBlock) {
     return transferableBlock.isEndOfStreamBlock();
   }
+
+  /**
+   *  Split a block into multiple block so that each block size is within maxBlockSize.
+   *  Currently, we only support split for row type dataBlock.
+   *  For columnar data block, we return the original data block.
+   *  Metadata data block split is not supported.
+   *
+   *  When row size is greater than maxBlockSize, we pack each row as a separate block.
+   */
+  public static List<BaseDataBlock> getDataBlockChunks(BaseDataBlock block, BaseDataBlock.Type type, int maxBlockSize)
+      throws IOException {
+    List<BaseDataBlock> blockChunks = new ArrayList<>();
+    switch (type) {
+      // TODO: Add support for columnar data block.
+      case COLUMNAR:
```

Comment:
should we fail here as well? I think it's confusing to have a function that doesn't do what it says.